examples/images_contours_and_fields/streamplot_demo.py

```diff
@@ -0,0 +1,68 @@
+"""
+==========
+Streamplot
+==========
+
+Demo of the `streamplot` function.
+
+A streamplot, or streamline plot, is used to display 2D vector fields. This
+example shows a few features of the stream plot function:
+
+    * Varying the color along a streamline.
+    * Varying the density of streamlines.
+    * Varying the line width along a stream line.
+    * Streamlines skipping masked regions and NaN values.
+    * Controlling the start points of streamlines.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+w = 3
+Y, X = np.mgrid[-w:w:100j, -w:w:100j]
+U = -1 - X**2 + Y
+V = 1 + X - Y**2
+speed = np.sqrt(U*U + V*V)
+
+# Varying color along a streamline
+fig0, ax = plt.subplots()
+strm = ax.streamplot(X, Y, U, V, color=U, linewidth=2, cmap='autumn')
+fig0.colorbar(strm.lines)
+
+# Varying the density of streamlines
+fig1, (ax1, ax2) = plt.subplots(ncols=2)
+ax1.streamplot(X, Y, U, V, density=[0.5, 1])
+
+# Varying the line width along a stream line
```

Comment:
I would merge this block with the previous one. I find it a bit perturbing that the previous block creates two subplots but only uses one. IMO, the two description comments may then be merged into something like:
```python
# Varying the density of streamlines (ax1) and the line width along
# a streamline (ax2).
```